### ISSUE TITLE
zipp.Path.glob does not find directories

### ISSUE THREAD

**Issue by theunkn0wn1** [jaraco/zipp#121]
using glob to find folders via zipp.Path.glob does not find the folder.
This issue also manifests in Cpython's `zipfile.Path` implementation, which appears to inherit from this codebase.

## Environment
Ubuntu 22.04 Focal Fossa
Python3.10, python3.12.3 (obtained via pyenv)
```
 name         : zipp                                                        
 version      : 3.19.2                                                      
 description  : Backport of pathlib-compatible object wrapper for zip files 
```                                                                                        
                                                                                        

## Minimal example
First, build a zip archive with sufficient structure
```bash
mkdir -p /tmp/something
cd /tmp/something
mkdir -p top_level-4242/the_child_2213-039/
echo "i exist!!!" > top_level-4242/the_child_2213-039/flag.txt
zip demo.zip top_level-4242/the_child_2213-039/flag.txt   
```

Second, prove that with `pathlib.Path.glob` the folder is detectable:
```py
from zipfile import ZipFile # to open the zip archive
from zipp import Path as ZipPath  # for the pathlib-compatible interface
from pathlib import Path  # for the pathlib interface
# path to target zip archive
target = Path("/tmp/something/zip_repo/demo.zip")
# path to source material on disk
as_pathlib = target.parent / "top_level-4242"
assert as_pathlib.exists() , "directory wasn't set up ahead of time!"
assert list(as_pathlib.glob("the_child*")), "pathlib: glob pattern miss."
# assertion passes
```
The assertion passes.

Third, demonstrate that `zipp.Path` does not behave as expected:

```py
... # see above for setup
assert target.exists(), "sample zip archive not set up in advance!!!"
root = ZipPath(ZipFile(target)).joinpath("top_level-4242")
assert(list(root.glob("the_child*"))), "zipp: glob pattern miss."
# assertion error

```

**Comment by jaraco** [comment]
I just came here to report the same issue! Thanks for reporting it.

**Comment by jaraco** [comment]
Fix released in v3.19.3.
